scripts/bootstrap.sh

```diff
@@ -1146,12 +1147,23 @@ install_gcc() {
 	execute_sudo ln -sf $(which llvm-symbolizer-$llvm_v) /usr/bin/llvm-symbolizer
 }
 
-install_ccache() {
-	case "$pm" in
-	apt | apk | brew)
-		install_packages ccache
-		;;
-	esac
+install_sccache() {
+    # Alright, look, this function is cobbled together but it's only as cobbled
+    # together as this whole script is.
+    #
+    # For some reason, move_to_bin doesn't work here due to permissions so I'm
+    # avoiding that function. It's also wrong with permissions and so on.
+    set -e
+    set -f
+
+    local sccache_http="https://github.com/mozilla/sccache/releases/download/v0.12.0/sccache-v0.12.0-$(uname -m)-unknown-linux-musl.tar.gz"
+    local file=$(download_file "$sccache_http")
+    local tmpdir=$(mktemp -d)
```

Comment:
```suggestion
    local file
    local tmpdir
    file=$(download_file "$sccache_http")
    tmpdir=$(mktemp -d)
```

On many shells, `local foo=$(cmd)` always succeeds, even if `cmd` errors, so `set -e` won't catch errors in these commands.

Reply:
TIL